static/js/services/enketo.js

```diff
@@ -63,50 +65,55 @@ angular.module('inboxServices').service('Enketo',
     };
 
     var transformXml = function(doc) {
-      return $q
-        .all([
-          XSLT.transform('openrosa2html5form.xsl', doc),
-          XSLT.transform('openrosa2xmlmodel.xsl', doc),
-        ])
-        .then(function(results) {
-          var result = {
-            html: $(results[0]),
-            model: results[1]
-          };
-          return result;
-        });
+      return $q.all([
+        XSLT.transform('openrosa2html5form.xsl', doc),
+        XSLT.transform('openrosa2xmlmodel.xsl', doc),
+      ])
+      .then(function(results) {
+        return {
+          html: $(results[0]),
+          model: results[1]
+        };
+      });
     };
 
-    var xmlCache = {};
+    var translateXml = function(text, title, language) {
+      var xml = $.parseXML(text);
+      var $xml = $(xml);
+      // set the user's language as default so it'll be used for itext translations
+      $xml.find('model itext translation[lang="' + language + '"]').attr('default', '');
+      // manually translate the title as itext doesn't seem to work
```

Comment:
Yeah absolutely. I'm trying to work out where to raise this. I'll merge this in the meantime.